### DISCUSSION TITLE
Have the camera streams issue after the lates update

### DISCUSSION THREAD

**Discussion by cstamopo** [tsightler/ring-mqtt#853] [Support Q&A]
Hi all,
After the latest update a few days ago I noticed that my cameras were not recording on motion events.
Trying to open the camera entities to see the live view was also failing.
I tried from VLC with the given stream link and was also failing.

The cameras were working fine on the ring app.
All the camera entities in HA and automations were working fine for many months prior to the update.

I revert to the back up prior to the ring-mqtt update to v5.6.5 and the streams started working again.
As a test I updtaed again the addon to 5.6.5 and the streams stoped working.
I have now reverted back to the previous version and they are working again.

I am aware of the issue with the disarm code that the latest version is resolving and as a workaround on that I specified a disarm code.

On the Release v5.6.5 commit I see the gortc version got updtaed to 1.9.3. Could that be the issue or do I need to change some configuration?

Forgot to mention I am using the HA addon on HAOS running on RPi 4.

Thanks for the wonderful project and support.
I though it's worth reporting as it might be a bug that affects more people.

Costas

**Comment by chiboule** [comment]
Hello,

Same behaviour here. I have 3 doorbells and streams are buggy since last  ringmqtt update. HAOS too, nuc

**Comment by funks81** [comment]
I have the same issue

**Comment by DougManton** [comment]
I downgraded go2rtc to 1.8.5 and the streams have been stable since.

**Comment by tsightler** [comment]
It seems that go2rtc v1.9.3 made a change in how to stop the exec process which causes some bad behavior.  In ring-mqtt, go2rtc is configured to run a script which uses mosquitto_pub/sub cli tools to send messages to ring-mqtt to start/stop the specific camera stream.  However, with current go2rtc version, when there are no more consumer for a stream, it forcefully stops the exec process via SIGKILL, instead of the previous versions which used SIGTERM for a normal process exit.

Normally, the script traps a SIGTERM or SIGINT attempt and sends the upstream message to let ring-mqtt know to close down the respective stream.  However, with the process being terminated with SIGKILL, it's no longer possible to trap the exit and perform this cleanup.  Hopefully this will be addressed in the upstream go2rtc build but, if not, there are other ways to address the issue from a code perspective.

For now, the easiest solution is to use the dev version where I've temporarily reverted back to go2rtc v1.8.5, although I believe this change was implemented in v1.9.3 specifically, so even v1.9.2 is probably fine.

**Reply by funks81** [reply]
Sorry to ask a really stupid question - but how do i install the dev version?

**Reply by tsightler** [reply]
Documentation is your friend as it varies slightly based on addon vs docker install, but, overall, it's simply a matter of setting branch config option to 'dev' and restarting.

**Reply by funks81** [reply]
Thanks - got it!!

**Reply by pietje63** [reply]
Thanks for the quick action! Unfortunately the addon version (also after switch to dev) does not appear to be updated yet, though I stop digging as I assume an update will follow at some moment :)

**Reply by tsightler** [reply]
Addon version doesn't change, it just downloads the latest dev versions during startup, that is all.  You can see the go2rtc version in the log.

**Reply by tsightler** [reply]
The problem discussed in this thread is not about the camera being not availajble, it's about the camera only working perhaps one time after a restart and then not working after that.  Other than the changed go2rtc version and 2 lines of code to address the alarm issue, there are no other changes with this and the previous version so you'll have to figure out what's wrong on your own.   You can always just revert to the prior version of the plugin.

**Comment by parjanos** [comment]
HI.
I use "Ring-MQTT with Video Streaming" to display the live stream of my Ring camera using a „generic camera“ on the dashboard of my Home Assistant. This has worked without any problems so far. Since the update to 5.6.5, I have also a problem with the connection of the generic camera to "Ring-MQTT with Video Streaming". It breaks after a short time and only reconnects for a few minutes after restarting "Ring-MQTT with Video Streaming". Can you also solve your problems for a few minutes by restarting the add-on?

**Reply by tsightler** [reply]
This thread already contains the answers you seek.

**Reply by parjanos** [reply]
Do you mean to set the branch config option to 'dev' and restart?

It seems to work. ;)

Thanks a lot!
